server/admin_management/api/app/routers/router_access_control_privilege.py

```diff
@@ -76,34 +76,36 @@ def create_access_control_privilege_many(
     )
 
     try:
-        audit_event_log.requesting_user = user_service.get_user_by_cognito_user_id(
-            requester.cognito_user_id
-        )
+        audit_event_log.requesting_user = requester
         audit_event_log.role = role_service.get_role_by_id(
             access_control_privilege_request.role_id
         )
         audit_event_log.application = audit_event_log.role.application
 
-        return access_control_privilege_service.create_access_control_privilege_many(
+        response = access_control_privilege_service.create_access_control_privilege_many(
             access_control_privilege_request, requester.cognito_user_id, target_user
         )
+        # get target user from database, so for existing user, we can get the cognito user id
+        audit_event_log.target_user = user_service.get_user_by_domain_and_guid(
+            access_control_privilege_request.user_type_code,
+            access_control_privilege_request.user_guid,
+        )
+
+        return  response
 
     except Exception as e:
         audit_event_log.event_outcome = AuditEventOutcome.FAIL
         audit_event_log.exception = e
         raise e
 
     finally:
-        audit_event_log.target_user = user_service.get_user_by_domain_and_guid(
-            access_control_privilege_request.user_type_code,
-            access_control_privilege_request.user_guid,
-        )
+        # if failed to get target user from database, use the information from request
         if audit_event_log.target_user is None:
-            audit_event_log.target_user = models.FamUser(
-                user_type_code=access_control_privilege_request.user_type_code,
-                user_name=access_control_privilege_request.user_name,
-                user_guid="unknown",
-                cognito_user_id="unknown",
+            audit_event_log.target_user = FamUser(
+                user_type_code=target_user.user_type_code,
+                user_name=target_user.user_name,
+                user_guid=target_user.user_guid,
+                cognito_user_id=target_user.cognito_user_id,
```

Comment:
I see, so you tried to use `**target_user` but the schema object it not something that can be destructed.
I wonder if `**target_user.model_dump_json()` would work... but it does not matter for now.

Reply:
yeah I actually tried, but the target_user schema is based on the requester schema, and it has a field "access_roles". I got an error while using the model_dump. "TypeError: 'access_roles' is an invalid keyword argument for FamUser".